src/hist.jl

```diff
@@ -237,3 +239,150 @@ fit{N}(::Type{Histogram},vs::NTuple{N,AbstractVector}, wv::WeightVec; closed::Sy
     closed = _check_closed_arg(closed,:fit)
     fit(Histogram, vs, wv, histrange(vs,nbins,closed); closed=closed)
 end
+
+
+# Get a suitable high-precision type for the norm of a histogram.
+@generated function norm_type{T, N, E}(h::Histogram{T, N, E})
+    args = [:( eltype(edges[$d]) ) for d = 1:N]
+    quote
+        edges = h.edges
+        norm_type(promote_type(T, $(args...)))
+    end
+end
+
+norm_type{T<:Integer}(::Type{T}) = promote_type(T, Int64)
+norm_type{T<:AbstractFloat}(::Type{T}) = promote_type(T, Float64)
+
+
+"""
+    norm(h::Histogram)
+
+Calculate the norm of histogram `h` as the absolute value of its integral.
+"""
+@generated function norm{T, N, E}(h::Histogram{T, N, E})
+    quote
+        edges = h.edges
+        weights = h.weights
+        S = norm_type(h)
+        v_0 = 1
+        s_0 = zero(S)
+        @inbounds @nloops(
+            $N, i, weights,
+            d -> begin
+                v_{$N-d+1} = v_{$N-d} * (edges[d][i_d + 1] - edges[d][i_d])
+                s_{$N-d+1} = zero(S)
+            end,
+            d -> begin
+                s_{$N-d} += s_{$N-d+1}
+            end,
+            begin
+                $(Symbol("s_$(N)")) += (@nref $N weights i) * $(Symbol("v_$N"))
+            end
+        )
+        norm(s_0)
```

Comment:
> IMO the difference among two histograms is not a histogram

I have to disagree - in practice, subtraction of histograms is quite common, esp. for experimental data. One example is spectroscopy: It's common practice to subtract a background histogram from a signal histogram. Afterwards, most bins are (usually) positive, but if a bin has few counts, the signal value can be lower than the background value in that bin, due to statistical fluctuations - resulting in a negative bin value. Certainly this isn't very clean, mathematically/statistically, but common practice nevertheless. Any yes, people often do normalize the result in various ways (of course the integral of the difference histogram is still positive, usually).

So maybe it would be best to output `s_0` instead of `norm(s_0)`. Then a negative norm value can act as a clear warning that something is wrong.